lib/customValidators.js

```diff
@@ -174,6 +174,70 @@ function validateOperationId(
   return true;
 }
 
+/**
+ * Validates if messageIds are duplicated in the document
+ *
+ * @private
+ * @param  {Object} parsedJSON parsed AsyncAPI document
+ * @param  {String} asyncapiYAMLorJSON AsyncAPI document in string
+ * @param  {String} initialFormat information of the document was originally JSON or YAML
+ * @returns {Boolean} true in case the document is valid, otherwise throws {@link ParserError}
+ */
+function validateMessageId(
+  parsedJSON,
+  asyncapiYAMLorJSON,
+  initialFormat,
+  operations
+) {
+  const chnls = parsedJSON.channels;
+  if (!chnls) return true;
+  const chnlsMap = new Map(Object.entries(chnls));
+  //it is a map of paths, the one that is a duplicate and the one that is duplicated
+  const duplicatedMessages = new Map();
+  //is is a 2-dimensional array that holds information with messageId value and its path
+  const allMessages = [];
+
+  const addDuplicateToMap = (msg, channelName, opName, oneOf = '') => {
+    const messageId = msg.messageId;
+    if (!messageId) return;
+
+    const messagePath = `${tilde(channelName)}/${opName}/message${oneOf}/messageId`;
+    const isMessageIdDuplicated = allMessages.find(v => v[0] === messageId);
+    if (!isMessageIdDuplicated)
+      return allMessages.push([messageId, messagePath]);
+
+    //isMessageIdDuplicated always holds one record and it is an array of paths, the one that is a duplicate and the one that is duplicated
+    duplicatedMessages.set(messagePath, isMessageIdDuplicated[1]);
+  };
+
+  chnlsMap.forEach((chnlObj, chnlName) => {
+    operations.forEach((opName) => {
+      const op = chnlObj[String(opName)];
+      if (op && op.message) {
+        if (op.message.oneOf) op.message.oneOf.forEach(msg => addDuplicateToMap(msg, chnlName, opName , '/oneOf'));
```

Comment:
```suggestion
        if (op.message.oneOf) op.message.oneOf.forEach((msg, index)=> addDuplicateToMap(msg, chnlName, opName , `/oneOf/${index}`));
```

Reply:
What's the use of `index` here? 🤔

Reply:
`message/oneOf` has a shape of array, so you need to clarify which element of array has potential problems :)